JavaScript/JavaScriptSDK/Context/PageVisitTimeManager.ts

```diff
@@ -0,0 +1,96 @@
+﻿
+module Microsoft.ApplicationInsights.Context {
+    "use strict";
+
+    /**
+     * Used to track page visit durations
+     */
+    export class PageVisitTimeManager {
+
+        private prevPageVisitDataKeyName = "prevPageVisitData";
+
+        /**
+         * Stops timing of current page (if exists) and starts timing for duration of visit to pageName
+         * @param pageName Name of page to begin timing visit duration
+         * @returns {PageVisitData} Page visit data (including duration) of pageName from last call to start or restart, if exists. Null if not. 
+         */
+        restartPageVisitTimer(pageName: string, pageUrl: string) {
+
+            var prevPageVisitData = this.stopPageVisitTimer();
+            this.startPageVisitTimer(pageName, pageUrl);
+
+            return prevPageVisitData;
+        }
+
+        /**
+         * Starts timing visit duration of pageName
+         * @param pageName 
+         * @returns {} 
+         */
+        startPageVisitTimer(pageName: string, pageUrl: string) {
+            if (this.checkSessionStorageSupported()) {
+                if (window.sessionStorage.getItem(this.prevPageVisitDataKeyName) != null) {
+                    throw new Error("Cannot call startPageVisit consecutively without first calling stopPageVisit");
+                }
+
+                var currPageVisitData = new PageVisitData(pageName, pageUrl);
+                var currPageVisitDataStr = JSON.stringify(currPageVisitData);
+                window.sessionStorage.setItem(this.prevPageVisitDataKeyName, currPageVisitDataStr);
```

Comment:
Instead of direct call to window.sessionStorage please use Util class. Look at https://github.com/Microsoft/ApplicationInsights-JS/blob/master/JavaScript/JavaScriptSDK/Util.ts#L30 and https://github.com/Microsoft/ApplicationInsights-JS/blob/master/JavaScript/JavaScriptSDK/Util.ts#L59